cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectAssemblyOperator.java

```diff
@@ -257,283 +220,96 @@ protected KafkaConnectStatus createStatus() {
-                        buildState.desiredImage = state.getMessage().trim();
-                        buildState.desiredBuildRevision = newBuildRevision;
-                        LOGGER.infoCr(reconciliation, "Build completed successfully. New image is {}.", buildState.desiredImage);
-                        return Future.succeededFuture();
-                    } else {
-                        ContainerStateTerminated state = pod.getStatus().getContainerStatuses().get(0).getState().getTerminated();
-                        LOGGER.warnCr(reconciliation, "Build failed with code {}: {}", state.getExitCode(), state.getMessage());
-                        return Future.failedFuture("The Kafka Connect build failed");
-                    }
-                })
-                .compose(i -> podOperator.reconcile(reconciliation, namespace, KafkaConnectResources.buildPodName(connectBuild.getCluster()), null))
-                .compose(ignore -> pfa.supportsS2I() ? buildConfigOperator.reconcile(reconciliation, namespace, KafkaConnectResources.buildConfigName(connectBuild.getCluster()), null) : Future.succeededFuture())
-                .mapEmpty();
-    }
-
-    /**
-     * Executes the Kafka Connect Build on OpenShift. Run only if needed because of changes to the Dockerfile or when
-     * triggered by annotation.
-     *
-     * @param reconciliation        The reconciliation
-     * @param namespace             Namespace where the Kafka Connect is deployed
-     * @param connectBuild          The KafkaConnectBuild model with the build definitions
-     * @param buildState            State object of the Kafka Connect build used to pass information around
-     * @param dockerfile            The generated Dockerfile
-     * @param newBuildRevision      New build revision (hash of the Dockerfile)
-     *
-     * @return                      Future which completes when the build is finished (or fails if it fails)
-     */
-    private Future<Void> openShiftBuild(Reconciliation reconciliation, String namespace, KafkaConnectBuild connectBuild, BuildState buildState, KafkaConnectDockerfile dockerfile, String newBuildRevision)   {
-        return buildConfigOperator.getAsync(namespace, KafkaConnectResources.buildConfigName(connectBuild.getCluster()))
-                .compose(buildConfig -> {
-                    if (buildConfig != null
-                            && buildConfig.getStatus() != null
-                            && buildConfig.getStatus().getLastVersion() != null) {
-                        Long lastVersion = buildConfig.getStatus().getLastVersion();
-                        return buildOperator.getAsync(namespace, KafkaConnectResources.buildName(connectBuild.getCluster(), lastVersion));
-                    } else {
-                        return Future.succeededFuture();
+    Future<BuildState> connectBuild(Reconciliation reconciliation, String namespace, String connectName, KafkaConnectBuild connectBuild) {
```

Comment:
Have you considered moving this to the `ConnectBuildOperator` as well and making the `BuildState` the interface between the `ConnectBuildOperator` and the `KafkaConnectAssemblyOperator`? It seems like lot of the things here would better belong there. But I'm not sure if it can be easily moved or if it depends on some things which have to be here.

Reply:
I've moved it over and moved some of the variables to class level.